src/opaque.rs

```diff
@@ -762,35 +726,27 @@ impl<CS: CipherSuite> ClientLogin<CS> {
             l2.beta,
             &self.blinding_factor,
         )?;
-        let h = Hkdf::<Sha256>::new(None, &password_derived_key);
-        let mut okm = [0u8; 3 * DERIVED_KEY_LEN];
-        h.expand(STR_ENVU, &mut okm)
-            .map_err(|_| InternalPakeError::HkdfError)?;
-        let encryption_key = &okm[..DERIVED_KEY_LEN];
-        let hmac_key = &okm[DERIVED_KEY_LEN..2 * DERIVED_KEY_LEN];
-        let kd_key = &okm[2 * DERIVED_KEY_LEN..];
-
-        let client_s_sk = Key::from_bytes(
-            &l2.envelope
-                .decrypt(&encryption_key, &hmac_key, &server_s_pk.to_arr())
-                .map_err(|e| match e {
-                    PakeError::DecryptionHmacError => PakeError::InvalidLoginError,
-                    err => err,
-                })?,
-        )?;
+
+        let (client_s_sk, export_key) = &l2
+            .envelope
+            .open(&password_derived_key, &server_s_pk.to_arr())
+            .map_err(|e| match e {
+                PakeError::SealOpenHmacError => PakeError::InvalidLoginError,
```

Comment:
nit: this makes me think `SealOpenHmacError` should be an `InternalPakeError` which would translate to `InvalidLoginError` at the `PakeError` level

Reply:
Converting all errors thrown from envelope.rs to be InternalPakeError instead of PakeError